dd-sdk-android-gradle-plugin/src/main/kotlin/com/datadog/gradle/plugin/DdNdkSymbolFileUploadTask.kt

```diff
@@ -0,0 +1,191 @@
+package com.datadog.gradle.plugin
+
+import com.android.build.gradle.api.ApplicationVariant
+import com.android.build.gradle.tasks.ExternalNativeBuildTask
+import com.android.builder.model.Version
+import com.datadog.gradle.plugin.internal.ApiKey
+import com.datadog.gradle.plugin.internal.Uploader
+import org.gradle.api.Project
+import org.gradle.api.file.ConfigurableFileCollection
+import org.gradle.api.file.DirectoryProperty
+import org.gradle.api.logging.Logging
+import org.gradle.api.model.ObjectFactory
+import org.gradle.api.provider.Provider
+import org.gradle.api.provider.ProviderFactory
+import org.gradle.api.tasks.InputFiles
+import org.gradle.api.tasks.TaskProvider
+import java.io.File
+import javax.inject.Inject
+import kotlin.reflect.full.memberProperties
+
+/**
+ * A Gradle task to upload NDK symbol files to Datadog servers.
+ */
+internal abstract class DdNdkSymbolFileUploadTask @Inject constructor(
+    objects: ObjectFactory,
+    providerFactory: ProviderFactory
+) : DdFileUploadTask(providerFactory) {
+
+    @get:InputFiles
+    val searchDirectories: ConfigurableFileCollection = objects.fileCollection()
+
+    init {
+        description =
+            "Uploads NDK symbol files to Datadog servers to perform native crash symbolication."
+    }
+
+    override fun getFilesList(): List<Uploader.UploadFileInfo> {
+        val files = mutableListOf<Uploader.UploadFileInfo>()
+
+        searchDirectories
+            .flatMap(this::findSoFiles)
+            .toSortedSet(compareBy { it.absolutePath })
+            .toSet()
```

Comment:
still the same comment I posted before https://github.com/DataDog/dd-sdk-android-gradle-plugin/pull/220#discussion_r1479404196, imo this can be simply `toSet` here.

```suggestion
            .toSet()
```